src/sentry/integrations/jira/utils/create_issue_schema_transformers.py

```diff
@@ -0,0 +1,133 @@
+from collections.abc import Callable, Iterable, Mapping
+from typing import Any, TypeVar
+
+from sentry.integrations.jira.models.create_issue_metadata import (
+    JIRA_CUSTOM_FIELD_TYPES,
+    JiraField,
+    JiraSchemaTypes,
+)
+from sentry.shared_integrations.exceptions import IntegrationFormError
+
+
+class JiraSchemaParseError(Exception):
+    pass
+
+
+def parse_number_field(num_str: Any) -> int | float:
+    try:
+        if isinstance(num_str, str) and "." in num_str:
+            return float(num_str)
+
+        return int(num_str)
+    except ValueError:
+        raise JiraSchemaParseError(f"Invalid number value provided for field: '{num_str}'")
+
+
+TransformerType = Mapping[str, Callable[[Any], Any]]
+
+T = TypeVar("T")
+
+
+def identity_transformer(input_val: T) -> T:
+    return input_val
+
+
+def id_obj_transformer(input_val: Any) -> dict[str, Any]:
+    return {"id": input_val}
+
+
+def get_type_transformer_mappings(user_id_field: str) -> TransformerType:
+    transformers = {
+        JiraSchemaTypes.user.value: lambda x: {user_id_field: x},
+        JiraSchemaTypes.issue_type.value: id_obj_transformer,
+        JiraSchemaTypes.option.value: lambda x: {"value": x},
+        JiraSchemaTypes.issue_link.value: lambda x: {"key": x},
+        JiraSchemaTypes.project.value: id_obj_transformer,
+        JiraSchemaTypes.number.value: parse_number_field,
+    }
+
+    return transformers
+
+
+def get_custom_field_transformer_mappings() -> TransformerType:
+    transformers = {
+        # JIRA_CUSTOM_FIELD_TYPES["select"]: identity_transformer,
```

Comment:
is this yet to be created? is there any downside to not having it in this PR too?

Reply:
Yep! The logic that displays `select` in the UI is currently busted, so I need to patch that before this field can be useful. I'll add TODO comments on these to make this less ambiguous.

As for `epic`, that's another display issue that I need to dig further into, but I can't get anything with type `epic` to appear in the issue-link dropdown 🤔